src/hooks/useCurrency.ts

```diff
@@ -90,8 +91,8 @@ export function useTokenFromMapOrNetwork(tokens: TokenMap, tokenAddress?: string
  * Returns null if token is loading or null was passed.
  * Returns undefined if tokenAddress is invalid or token does not exist.
  */
-export function useToken(tokenAddress?: string | null): Token | null | undefined {
-  const tokens = useTokenMap()
+export function useToken(tokenAddress?: string | null, chainId?: SupportedChainId): Token | null | undefined {
+  const tokens = useTokenMap(chainId)
   return useTokenFromMapOrNetwork(tokens, tokenAddress)
```

Comment:
If there is a `chainId` passed, and `chainId !== activeChainId`, the network should not be queried.
`useTokenFromMapOrNetwork` does not use chainId, so I would pass this as an extra boolean parameter: `return useTokenFromMapOrNetwork(tokens, tokenAddress, /*skipNetwork=*/ true)`.

    export function useTokenFromMapOrNetwork(tokens: TokenMap, tokenAddress?: string | null, skipNetwork = false): ... {

Reply:
Good call I changed this. Unfortunately, I had to use `useWeb3React` inside `useToken` to get`activeChainId`